File: backend/src/izakaya_api/schemas/connector.py
from datetime import datetime

from pydantic import BaseModel


class ConnectorCreateRequest(BaseModel):
    name: str
    service: str


class ConnectorCreateResponse(BaseModel):
    id: int
    name: str
    connect_card_url: str


class ConnectorFinalize(BaseModel):
    fivetran_connector_id: str


class ConnectorUpdate(BaseModel):
    name: str | None = None
    status: str | None = None
    setup_state: str | None = None
    sync_state: str | None = None


class ConnectorResponse(BaseModel):
    id: int
    name: str
    fivetran_connector_id: str | None
    service: str
    status: str
    setup_state: str
    sync_state: str | None
    schema_name: str
    succeeded_at: datetime | None = None
    failed_at: datetime | None = None
    sync_frequency: int | None = None
    schedule_type: str | None = None
    paused: bool = False
    daily_sync_time: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/connector.ts
export interface Connector {
  id: number;
  name: string;
  fivetran_connector_id: string | null;
  service: string;
  status: string;
  setup_state: string;
  sync_state: string | null;
  succeeded_at: string | null;
  failed_at: string | null;
  sync_frequency: number | null;
  schedule_type: string | null;
  paused: boolean;
  daily_sync_time: string | null;
  connector_category: string;
  requires_table_selection: boolean;
  created_at: string;
  updated_at: string;
}

export interface ConnectorSummary {
  id: number;
  name: string;
  service: string;
  status: string;
  paused: boolean;
  succeeded_at: string | null;
  failed_at: string | null;
  sync_state: string | null;
  created_at: string;
}

export interface ConnectorType {
  id: string;
  name: string;
}

export interface ConnectorOption {
  id: number;
  name: string;
  schema_name: string;
  status: string;
  requires_table_selection: boolean;
  connector_category: string;
}

export interface BqTable {
  table_id: string;
  full_id: string;
}
